libgit/repo.go

```diff
@@ -238,7 +271,7 @@ func getOrCreateRepoAndID(
 		return nil, NullID, err
 	}
 
-	if createOnly {
+	if op == createOnly {
```

Comment:
How are we locking this?

Reply:
I don't think what "this" means here.  If we get to this line of code, we know for sure the repo existed at the time the request started.  That's enough to fail the request.  If this is being executed concurrently with a delete of the same repo, oh well, the user will just have to retry.  Does that make sense?

Reply:
Yep, thanks.